### PR TITLE
Reorganizes OceanProblems module and adds InitialValueProblem

### PR DESCRIPTION
**Pull request by glwagner**
### Description

This PR adds `InitialValueProblem <: AbstractSimpleBoxProblem` and subsidiary type `InitialConditions`.

`InitialConditions` is a container for initial conditions functions of `x, y, z`.

Usage looks like:

```julia
a = 0.01 # m
L = 1e5 # m

ηᵢ(x, y, z) = a * exp(-x^2 / 2L^2)

initial_conditions = InitialConditions(η=ηᵢ)

problem = InitialValueProblem(dimensions = (Lx, Ly, Lz), initial_conditions = initial_conditions)
```

`problem` is then passed to `HydrostaticBoussinesqModel`, which is one of the `BalanceLaw`s implemented in `ClimateMachine.Ocean`.

`InitialValueProblem` is a convenience type intended to be a temporary addition to the code that make it slightly easier to write simple tutorials and facilitate experimentation. It may be refactored in the near future, along with much of the interface associated with `HydrostaticBoussinesqModel`.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Ocean/OceanProblems/OceanProblems.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+HBModel = HydrostaticBoussinesqModel
+SWModel = ShallowWaterModel
+
+include("simple_box_problem.jl")
+include("homogeneous_box.jl")
+
+include("shallow_water_initial_states.jl")
+
+function ocean_init_state!(m::SWModel, p::HomogeneousBox, Q, A, localgeo, t)
```
Comment (kpamnany) [inline src/Ocean/OceanProblems/OceanProblems.jl:30]: In Atmos, we're ordering the parameters as `(problem, balance_law, Q, ...)` -- suggest we use the same ordering.

Comment (glwagner) [reply src/Ocean/OceanProblems/OceanProblems.jl:30]: Ah, thanks! This PR didn't touch these function signatures (just moved the function definitions around), so I'll open an issue to address this.

**Comment by glwagner** [comment]
Doesn't look like there is much Ocean documentation --- should we start now, or wait? I think it may help to reorganize the docs first eg #1715 .

**Comment by christophernhill** [comment]
> Doesn't look like there is much Ocean documentation --- should we start now, or wait? I think it may help to reorganize the docs first eg #1715 .

I think starting soon would be good - with an outline from resolving #1715 discussion.

**Comment by glwagner** [comment]
bors try

**Comment by christophernhill** [comment]
> 🔒 Permission denied
> 
> Existing reviewers: [click here to make glwagner a reviewer](https://app.bors.tech/repositories/11521/add-reviewer/glwagner)

@glwagner I think I just added you to reviewers so I think this should work if you try again.

**Comment by glwagner** [comment]
`bors` is pretty real. no filter

**Review by christophernhill — APPROVED** [review]

**Comment by christophernhill** [comment]
@glwagner is there any reason to not merge/close this PR #1703 and PR #1723 into ocean/dev?

**Comment by glwagner** [comment]
Err... this PR is obselete. Superceded by #1716 .

**Pull request closed without merging**